Dynamic sound loading

Phoenix-Starlight opened the pull request:
Implements dynamic sound (un)loading. Why? Because sound buffers are never unloaded and forever persist until exit or some trigger to release the buffers (reload).

embeddedt commented on common/src/main/java/org/embeddedt/modernfix/common/mixin/perf/dynamic_resources/SoundBufferLibraryMixin.java:
```diff
@@ -0,0 +1,47 @@
+        K k = notification.getKey();
+        if(k == null)
+            return;
+        ModernFix.LOGGER.warn("Evicted sound {}", k);
```
Any reason why this isn't tied to `debugDynamicSoundLoading` being enabled?

Phoenix-Starlight replied:
It early returns but I suppose I can wrap it into an else. Refcatoring again.

embeddedt replied:
```suggestion
        if(debugDynamicSoundLoading)
            ModernFix.LOGGER.warn("Evicted sound {}", k);
```

Phoenix-Starlight replied:
*facepalm* oops.

Phoenix-Starlight replied:
Wait, no I can just wrap everything into the if except the sound close method.

Phoenix-Starlight replied:
Force-pushed.

Phoenix-Starlight committed:
Dynamic sound loading

embeddedt reviewed (approved):
I'll do some testing before merging it but looks reasonable.

Phoenix-Starlight committed:
Split dynamic sounds into its own folder

embeddedt commented:
Had another thought: is `maximumSize` needed? Any new access to the cache will trigger removal of any expired entries, so we shouldn't need to cap the size ourselves.

Phoenix-Starlight commented:
I was not aware that access would trigger removal, I was only
considering the pathological case where it tries to load more than 255 sounds (java channel limit).

Phoenix-Starlight committed:
Remove entry limit
If limit+1 sounds get loaded the buffer deletion can fail due to it being bound to a source but not played yet. Also fixes a leak.
